### PR TITLE
added nulls check to solve NPE for WINDUP-789

### PR DESCRIPTION
**Pull request by mareknovotny**

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverRmiRuleProvider.java [hunk 1]
```diff
@@ -85,9 +85,12 @@ private void extractMetadata(GraphRewrite event, JavaTypeReferenceModel typeRefe
 
             // Create the source report for the RMI Implementation.
             JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
-            for (JavaSourceFileModel source : javaClassService.getJavaSource(rmiServiceModel.getImplementationClass().getQualifiedName()))
-            {
-                source.setGenerateSourceReport(true);
+            
+            if (rmiServiceModel != null && rmiServiceModel.getImplementationClass() != null) {
```
Comment (jsight) [inline rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverRmiRuleProvider.java]: Technically the first rmiServiceModel null check is unnecessary here, as it is always initalized by the getOrCreate call above. It isn't harmful, though.

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

**Comment by jsight** [comment]
@windupgithubbot1 retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/473/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1104/
Test PASSed.

**Pull request merged**